internal/daemon/worker/session/manager.go

```diff
@@ -0,0 +1,110 @@
+package session
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	pbs "github.com/hashicorp/boundary/internal/gen/controller/servers/services"
+)
+
+// Manager stores session information locally and exposes ways to operate on the
+// set of sessions locally in batch.
+// This is thread-safe.
+type Manager struct {
```

Comment:
Sorry, I muddled the message. I didn't mean thread-safe, I just meant safe. If they instantiate the manager directly instead of with the constructor, the map and the client are never initialized, so unless you wrap all accesses to them with nil checks, you can't guarantee that you won't panic.

Which reminds me -- there should be an IsNil check on the passed-in client and the constructor should return an error if it's nil :-)